content/docs/tasks/traffic_management/envoy.md

```diff
@@ -0,0 +1,298 @@
+# Life of an OSM Request
```

Comment:
This doesn't pertain to traffic management from a configuration perspective. I know I suggested this as a potential location, but this folder only houses user-facing feature documentation, and looks out of place here.

@bridgetkromhout, @zr-msft what's the best place to put internal OSM implementation details without distracting users with information not relevant to configuring OSM?

Reply:
@shashankram would internal implementation details be more relevant to developers contributing to OSM? In that case, instead of on the user-facing website, I might suggest https://github.com/openservicemesh/osm/tree/main/docs as a good place for such docs.

Reply:
The thing is this is a good document with information on how traffic gets routed through OSM. It just seems slightly out of place with the other docs in this folder primarily pertaining to user-facing features. This isn't a feature doc, but rather informational describing internal details. It could be applicable to users but depends on if they really need to know about this. That's where I'd like your feedback @bridgetkromhout.

Reply:
I feel like internal details should remain in the main osm repo with the code as opposed to the main docs. 

If content is primarily targeted at end-users to use the product, it should go in the main docs.
If content is primarily targeted at contributors or users extending OSM, it should stay with the source code.

Reply:
I have read through the document and I agree that it is a good document with information about exactly how the code _works_. I agree that it should live in the code repo so as to inform future implementations. I think if end users need these details we should surface them in config options or in feature flags, such that there is something the end user can _do_. @shashankram is correct in saying that this doc is out of place in a folder devoted to user-facing features.

So I think the move here is, we put this document in the code repo. Do we all believe that is its best destination? Thanks.